src/bz2.cpp

```diff
@@ -55,15 +59,16 @@ void check_bz2_error(int e, const std::string& name)
     throw std::runtime_error(name + " failed: unknown error!");
 }
 
-std::string compress(std::string s, bool* compressed)
+std::vector<char> compress(const std::vector<char>& v, bool* compressed)
 {
-    std::string result = s;
-    unsigned int len   = result.size();
-    auto e             = BZ2_bzBuffToBuffCompress(&result[0], &len, &s[0], s.size(), 9, 0, 30);
+    auto result      = v;
+    unsigned int len = result.size();
+    auto e           = BZ2_bzBuffToBuffCompress(
+        result.data(), &len, const_cast<char*>(v.data()), v.size(), 9, 0, 30);
```

Comment:
@apwojcik HIP tidy is complaining about this line  
error: do not use const_cast [cppcoreguidelines-pro-type-const-cast,-warnings-as-errors]

Reply:
I guess HIP clang-tidy ignores fully-qualified exception `// NOLINTBEGIN(cppcoreguidelines-pro-type-const-cast)`
For me only global exception works, just `// NOLINTBEGIN`.